### DISCUSSION TITLE
Problem with sensor "name's" in H.A.

### DISCUSSION THREAD

**Discussion by MarCiu11** [dawidchyrzynski/arduino-home-assistant#91] [General]
Recently, I have a problem with the names of sensors in H.A.
I define a sensor for example:
`HASensor temp("Temp");`

in setup(){}
`temp.setUnitOfMeasurement("°C");`
` temp.componentName("Temp");`

And send to H.A.:
`temp.setValue(s);`

Home Assistant can see the sensor and its value without a problem, but its name is "**MQTT Sensor**" and so is the case with every sensor.
Earlier it was ok. Now I wanted to make a new thermometer and the reading of temperature, pressure and humidity looks like "**MQTT Sensor**"

**Comment by dawidchyrzynski** [comment]
Hi @MarCiu11,

There is "setName" method available that allows setting HA label. Based on your example it should be:

```cpp
temp.setUnitOfMeasurement("°C");
temp.setName("Temp");
```
